optionlab/engine.py

```diff
@@ -514,226 +271,280 @@ def pl_to_csv(self, filename="pl.csv", leg=-1):
+        ) and self._s_mc.shape[0] > 0:
+            profit = self.strategy_profit_mc[self.strategy_profit_mc >= 0.01]
+            loss = self.strategy_profit_mc[self.strategy_profit_mc < 0.0]
+            optional_outputs["average_profit_from_mc"] = 0.0
+            optional_outputs["average_loss_from_mc"] = (
+                loss.mean() if loss.shape[0] > 0 else 0.0
             )
 
+            if profit.shape[0] > 0:
+                optional_outputs["average_profit_from_mc"] = profit.mean()
+
+            if loss.shape[0] > 0:
+                optional_outputs["average_loss_from_mc"] = loss.mean()
+
+            optional_outputs["probability_of_profit_from_mc"] = (
+                self.strategy_profit_mc >= 0.01
+            ).sum() / self.strategy_profit_mc.shape[0]
+
+        return Outputs.model_validate(
+            optional_outputs
+            | {
+                "probability_of_profit": self.project_probability,
+                "strategy_cost": sum(self.cost),
+                "per_leg_cost": self.cost,
+                "profit_ranges": self._profit_ranges,
+                "minimum_return_in_the_domain": self.strategy_profit.min(),
+                "maximum_return_in_the_domain": self.strategy_profit.max(),
+                "implied_volatility": self.implied_volatility,
+                "in_the_money_probability": self.itm_probability,
+                "delta": self.delta,
+                "gamma": self.gamma,
+                "theta": self.theta,
+                "vega": self.vega,
+            }
+        )
+
     """
     Properties
     ----------
-    days2target : int, readonly
-        Number of days remaining to the target date from the start date.
-    stockpricearray : array
+    stock_price_array : array
         A Numpy array of consecutive stock prices, from the minimum price up to 
         the maximum price in the stock price domain. It is used to compute the 
         strategy's P/L profile.
-    terminalstockprices : array
+    terminal_stock_prices : array
         A Numpy array or terminal stock prices typically generated by Monte Carlo 
         simulations. It is used to compute strategy's expected profit and loss. 
     """
 
     @property
-    def days2target(self):
-        return self.__days2target
-
-    @property
-    def stockpricearray(self):
-        return self.__s
-
-    @stockpricearray.setter
-    def stockpricearray(self, s):
-        if isinstance(s, ndarray):
-            if s.shape[0] > 0:
-                self.__s = s
-            else:
-                raise ValueError("Empty stock price array is not allowed!")
-        else:
-            raise TypeError("A numpy array is expected!")
+    def stock_price_array(self):
+        return self.s
 
     @property
-    def terminalstockprices(self):
-        return self.__s_mc
+    def terminal_stock_prices(
+        self,
+    ):  # FIXME: why are Monte Carlo stock prices named `terminal_stock_prices` here?
+        return self._s_mc
```

Comment:
i guess the obvious question here is why not just call `_s_mc` `terminal_stock_prices and remove the property/setter @rgaveiga

Reply:
'terminal_stock_prices' instead of 's_mc' has a clearer meaning. Indeed, I think that a more descriptive name could enhance readability further. Thinking better about it, I would make this a readonly property, without a setter function. To avoid recreating the array of prices with each call to 'run()', caching could be implemented for the Monte Carlo function(s).